shared-module/vectorio/Polygon.c

```diff
@@ -56,22 +58,35 @@ static void _clobber_points_list(vectorio_polygon_t *self, mp_obj_t points_tuple
 
 
 void common_hal_vectorio_polygon_construct(vectorio_polygon_t *self, mp_obj_t points_list) {
-    VECTORIO_POLYGON_DEBUG("%p polygon_construct\n", self);
+    VECTORIO_POLYGON_DEBUG("%p polygon_construct: ", self);
     self->points_list = NULL;
     self->len = 0;
     self->on_dirty.obj = NULL;
     _clobber_points_list( self, points_list );
+    VECTORIO_POLYGON_DEBUG("\n");
 }
 
 
 mp_obj_t common_hal_vectorio_polygon_get_points(vectorio_polygon_t *self) {
-    return self->points_list;
+    VECTORIO_POLYGON_DEBUG("%p common_hal_vectorio_polygon_get_points {len: %d, points_list: %p}\n", self, self->len, self->points_list);
+    mp_obj_t list = mp_obj_new_list(0, NULL);
```

Comment:
I believe that if you call `mp_obj_new_list(self->len/2, NULL);` then the list will be initially allocated with the correct size, avoiding excess allocations.

Reply:
Sure, that's better.